lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -29,12 +29,25 @@ class NormalStatementHandler < StatementHandler
     # @param db [Sequel::Database]
     # @param sql_last_value [Integet|DateTime|Time]
     # @yieldparam row [Hash{Symbol=>Object}]
-    def perform_query(db, sql_last_value, jdbc_paging_enabled, jdbc_page_size)
+    def perform_query(db, sql_last_value, jdbc_paging_enabled, jdbc_paging_avoid_count, jdbc_page_size)
       query = build_query(db, sql_last_value)
       if jdbc_paging_enabled
-        query.each_page(jdbc_page_size) do |paged_dataset|
-          paged_dataset.each do |row|
-            yield row
+        unless jdbc_paging_avoid_count
+          query.each_page(jdbc_page_size) do |paged_dataset|
```

Comment:
I found 
```ruby
unless 
  #code segment A
else 
  #code segment B
end
```
a little bit counterintuitive

I would use the normal `if` and just swapping the code segments